### PR TITLE
new protease digestion motifs

### PR DESCRIPTION
**Pull request by bintiibrahim**

### CODE REVIEW

File: Proteomics/ProteolyticDigestion/Protease.cs [hunk 1]
```diff
@@ -172,21 +170,84 @@ public CleavageSpecificity GetCleavageSpecificity(string proteinSequence, int st
         internal List<int> GetDigestionSiteIndices(string proteinSequence)
         {
             var indices = new List<int>();
-            for (int i = 0; i < proteinSequence.Length - 1; i++)
+            string proteinSubstring = proteinSequence;
```
Comment (rmillikin) [inline Proteomics/ProteolyticDigestion/Protease.cs]: I'd like to see a lot of this code moved into DigestionMotif (I've added a few shells of methods that need completing there). Right now this class is kind of hard for me to follow. If we break this up into smaller methods in more appropriate places I think it'll be easier to understand

File: Test/DoubleProtease.tsv [hunk 2]
```diff
@@ -1,4 +1,4 @@
 Name	Sequences Inducing Cleavage	Sequences Preventing Cleavage	Cleavage Terminus	Cleavage Specificity	PSI-MS Accession Number	PSI-MS Name	Site Regular Expression	Notes
-Test1	K|R|D	"P,D|P|"	C|N|N	full				
-Test2	K|D		C|C	full				
-Test3	K|		C|N	full				
+Test1	"K|,|R,|D[D]"			full
```
Comment (rmillikin) [inline Test/DoubleProtease.tsv]: is this the same protease as before? shouldn't it be
`K|[P],K|[D],|R[P],|D` ?

Comment (bintiibrahim) [reply Test/DoubleProtease.tsv]: so i took a look at the test and the protein OKAREDY would be digested into: OK, A, REDY under Test 1.
so maybe the protease would be K|[P],K|[D],|R[P],|D**[D]** ?

File: Proteomics/ProteolyticDigestion/Protease.cs [hunk 3]
```diff
@@ -172,21 +167,19 @@ public CleavageSpecificity GetCleavageSpecificity(string proteinSequence, int st
         internal List<int> GetDigestionSiteIndices(string proteinSequence)
         {
             var indices = new List<int>();
-            for (int i = 0; i < proteinSequence.Length - 1; i++)
+            for (int i = 0; i < proteinSequence.Length; ++i)
             {
-                foreach (var c in SequencesInducingCleavage)
+                foreach(var c in MotifList)
```
Comment (rmillikin) [inline Proteomics/ProteolyticDigestion/Protease.cs]: much cleaner. thx. this looks good. I have some minor code style comments we should talk about but this is good

Comment (rmillikin) [reply Proteomics/ProteolyticDigestion/Protease.cs]: also need to redo the benchmarks for time and number of peptides

Comment (bintiibrahim) [reply Proteomics/ProteolyticDigestion/Protease.cs]: I did the benchmark test yesterday for 4 different protease and it all looks good

Comment (rmillikin) [reply Proteomics/ProteolyticDigestion/Protease.cs]: awesome 👍

**Pull request merged**